addons/stock/migrations/12.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2018 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_column_renames = {
+    'stock_location_path': [
+        ('location_from_id', 'location_src_id'),
+        ('location_dest_id', 'location_id'),
+    ],
+    'stock_move':
+    [
+        ('push_rule_id', None),
+    ],
+}
+
+_model_renames = [
+    ('procurement.rule', 'stock.rule'),
+]
+
+_table_renames = [
+    ('procurement_rule', 'stock_rule'),
+]
+
+
+@openupgrade.migrate(use_env=False)
+def migrate(cr, version):
+    openupgrade.rename_columns(cr, _column_renames)
+    openupgrade.rename_models(cr, _model_renames)
+    openupgrade.rename_tables(cr, _table_renames)
+    cr.execute(
+        """
+        ALTER TABLE stock_rule
+        ADD COLUMN {} integer;
+        """.format(openupgrade.get_legacy_name('loc_path_id'))
```

Comment:
I usually used `orig_` prefix for indicating that this comes from an insert operation and this is the reference of the original record

Reply:
well, but if we are using openupgrade to perform this changes then I prefer to use the get_legacy_name here